BasicMusicPlayer/app/src/main/java/data/artwork/PlaylistImager.kt

```diff
@@ -1,43 +1,223 @@
+                        playlist[i].playcut.imageURL = discogsURL
+                        continue
+                    }
+                    // if all release searches fail, use artist image
+                    val artistUrl = fetchDiscogsArtistImage(playlist[i])
+                    if (artistUrl != null) {
+                        playlist[i].playcut.imageURL = artistUrl
+                        println("artist image filled in")
+                        continue
+                    }
+                    println("All fetchers are null for " + playlist[i].playcut)
+                }
+            }
+            return@coroutineScope deferred
+        }.await()
 
-    suspend fun fetchImage(playcut: PlaylistDetails) {
+    // fetches the image url from discogs
+    suspend fun fetchDiscogsImage(playcut: PlaylistDetails): String? {
         try {
-
             val artist = playcut.playcut.artistName
-            val release = playcut.playcut.releaseTitle
-
-
+            var release = playcut.playcut.releaseTitle
+            if (release.equals("s/t", ignoreCase = true)) {
+                release = playcut.playcut.artistName
+            }
             val response = DiscogsArtFetcher.discogsService.getImage(
                 artist,
                 release,
-                "tYvsaskeJxOQbWoZSSkh",
-                "vZuPZFFDerXIPrBfSNnNyDhXjpIUiyXi"
+                DISCOGS_API_KEY,
+                DISCOGS_SECRET_KEY
             )
             if (response.isSuccessful) {
-                //println("FETCH IMAGE CHECKPOINT 2")
                 val responseBody = response.body()
                 if (responseBody != null) {
                     val discogsJson = responseBody.string()
                     val searchResult = Gson().fromJson(discogsJson, DiscogsResults::class.java)
+                    var imageUrl: String? = null
+                    for (result in searchResult.results) {
+                        if (result.coverImage?.endsWith(".gif") == false) {
+                            imageUrl = result.coverImage
+                            break
+                        }
+                    }
+                    if (imageUrl == "") {
+                        return null
+                    }
+                    if (imageUrl != null) {
+                        return imageUrl
+                    }
+                } else {
+                    // Handle empty response body
+                    println("discogs empty image data")
+                }
+            } else {
+                // Handle the error response
+                println("discogs art fetcher failed")
+            }
+        } catch (e: Exception) {
+            // Handle network or other exceptions
+        }
+        return null
+    }
 
-                    val imageUrl = searchResult.results.firstOrNull()?.thumb
+    //fetches the image url from itunes
+    suspend fun fetchItunesImage(playcut: PlaylistDetails): String? {
```

Comment:
Maybe name this `fetchItunesImageURL` since it's not actually fetching the image?